### PR TITLE
CTC-98 Add shadow to OakMediaClipList accordion

### PR DESCRIPTION
**Pull request by weronika-szalas**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below
Added shadow to `OakMediaClipList` component:
- shadow shouldn't appear if there is no scroll
- shadow should disappear when the list is scrolled to the end

## Link to the design doc
https://www.figma.com/design/iTimVTBZ3XcszhRa9KtD7r/Cycle-2-requirements---Teachers?node-id=0-1&node-type=canvas&focus-id=11-14616&m=dev

## A link to the component in the deployment preview
https://deploy-preview-335--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-teacher-oakmediacliplist--docs

### CODE REVIEW

**Comment by RBrownsett** [comment]
Tested, no issues. Will be checked again when merged into to OWA

File: src/components/molecules/InternalChevronAccordion/InternalChevronAccordion.tsx [hunk 1]
```diff
@@ -88,6 +109,36 @@ const Accordion = ({
   id,
   ...styleProps
 }: InternalChevronAccordionProps) => {
+  const [shouldDisplayShadow, setShouldDisplayShadow] = useState(false);
+  const scrollBox = useRef(null as null | HTMLDivElement);
```
Comment (k-huggs) [inline src/components/molecules/InternalChevronAccordion/InternalChevronAccordion.tsx:113]: interesting why does null need the type assertion?

**Review by k-huggs — APPROVED** [review]

File: src/components/molecules/InternalChevronAccordion/InternalChevronAccordion.tsx (hunk 1 above)
Comment (weronika-szalas) [reply src/components/molecules/InternalChevronAccordion/InternalChevronAccordion.tsx]: I was getting type errors but actually it should be this:
`const scrollBox = useRef<HTMLDivElement>(null);`

updated

**Pull request merged**